src/util/MergingUtil.js

```diff
@@ -0,0 +1,138 @@
+        shape1V.splice(shape1V.length - 1, 1);
+        shape2V.splice(shape2V.length - 1, 1);
+
+        let shape1Map = this.markSharedVertices(shape1V, shape2V);
+        let shape2Map = this.markSharedVertices(shape2V, shape1V);
+     
+        shape1V = this.removeInteriorVertices(shape1V, shape1Map);
+        shape2V = this.removeInteriorVertices(shape2V, shape2Map);
+
+        let mergedVertices = this.mergeVertices(shape1V, shape2V, shape1Map, shape2Map);
+
+        let shape1Neighbors = shape1.neighbors;
+        let shape2Neighbors = shape2.neighbors;
+        let mergedNeighbors = shape1Neighbors.concat(shape2Neighbors.filter((neighbor) => shape1Neighbors.indexOf(neighbor) < 0));
+    
+        mergedNeighbors.splice(mergedNeighbors.indexOf(shape2.seed), 1);
+        mergedNeighbors.splice(mergedNeighbors.indexOf(shape1.seed), 1);
+        let mergedShape = new ColorShape("black", mergedVertices, mergedNeighbors, shape1.seed);
+        
+        return mergedShape;
+    }
+
+    static markSharedVertices(shape1Vertices, shape2Vertices) {
+        let sharedVertexMap = new Map();
+        for(const vertex of shape1Vertices) {
+            if (this.isVertexIn(shape2Vertices, vertex)) {
+                sharedVertexMap.set(vertex, true);
+            } else {
+                sharedVertexMap.set(vertex, false);
+            }
+        }
+        return sharedVertexMap;
+    }
+
+    static isVertexIn(array, vertex) {
+        for(let v of array) {
+            if(vertex[0] == v[0] && vertex[1] == v[1]) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    static indexOfVertex(array, vertex) {
+        for(let i = 0; i < array.length; i++) {
+            if (array[i][0] == vertex[0] && array[i][1] == vertex[1]) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    static removeInteriorVertices(shapeVertices, shapeMap) {
+        for (let i = 0; i < shapeVertices.length; i++) {
+            let beforeIndex = i - 1;
+            if (beforeIndex == -1) {
+                 beforeIndex += shapeVertices.length;
+            }
+ 
+            let afterIndex = i + 1;
+            if (afterIndex == shapeVertices.length) {
+                afterIndex = 0;
+            }
+ 
+            let beforeIsShared = shapeMap.get(shapeVertices[beforeIndex]);
+            let afterIsShared = shapeMap.get(shapeVertices[afterIndex]);
+ 
+            if (beforeIsShared && afterIsShared) {
+                shapeVertices = shapeVertices.splice(i, 1);
+                i--;
+            }
+        }
+        return shapeVertices;
+    }
+
+    static mergeVertices(shape1Vertices, shape2Vertices, shape1Map, shape2Map) {
+        let mergedVertices = [];
+        let shape1Index = 0;
+        let shape2Index = 0;
+        let shapeSelected = 1;
+        let start = true;
+
+        while(true) {
+            if (shapeSelected == 1) {
```

Comment:
Its good practice in JavaScript (since all the variables are so volatile) to use a `===` instead of `==` since this also ensures the **type** of these objects are equal.